samples/insert-query-with-get-trasanction.js

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// sample-metadata:
+//  title: Perform a dml insert query using getTransaction method with optimisticLock
+//  usage: node insert-query-with-get-transaction.js <INSTANCE_ID> <DATABASE_ID> <PROJECT_ID>
+
+async function main(instanceId, databaseId, projectId) {
+  // [START spanner_insert_query_with_get_transaction]
+  // Imports the Google Cloud client library.
+  const {Spanner} = require('@google-cloud/spanner');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = 'my-project-id';
+  // const instanceId = 'my-instance';
+  // const databaseId = 'my-database';
+
+  // Creates a client
+  const spanner = new Spanner({
+    projectId: projectId,
+  });
+
+  async function writeTransaction() {
+    // Gets a reference to a Cloud Spanner instance and database
+    const instance = spanner.instance(instanceId);
+    const database = instance.database(databaseId);
+
+    const options = {
+      optimisticLock: true,
+    };
+
+    const promise = await database.getTransaction(options).then();
+    const transaction = promise[0];
+
+    try {
+      const [rowCount] = await transaction.runUpdate({
+        sql: `INSERT Singers (SingerId, FirstName, LastName) VALUES
+                    (1, 'Melissa', 'Garcia'),
+                    (2, 'Russell', 'Morales'),
+                    (3, 'Jacqueline', 'Long'),
+                    (4, 'Dylan', 'Shaw')`,
+      });
+      console.log(
+        `Successfully inserted ${rowCount} records into the Singers table`
+      );
+    } catch (err) {
+      console.error('ERROR:', err);
+    }
+
+    await transaction.commit();
```

Comment:
It seems strange to me that the sample commits the transaction regardless whether the insert statement succeeded or not. Put another way: Would it not make more sense to put the `commit()` call inside the try block? And then also add a `finally` block and put the `transaction.end()` call there.

Reply:
Done